test/runtime_test.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright 2016-2019 Envoy Project Authors
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "include/proxy-wasm/context.h"
+#include "include/proxy-wasm/wasm.h"
+
+#if defined(WASM_V8)
+#include "include/proxy-wasm/v8.h"
+#endif
+#if defined(WASM_WAVM)
+#include "include/proxy-wasm/wavm.h"
+#endif
+#if defined(WASM_WASMTIME)
+#include "include/proxy-wasm/wasmtime.h"
+#endif
+
+namespace proxy_wasm {
+namespace {
+
+struct DummyIntegration : public WasmVmIntegration {
+  ~DummyIntegration() override{};
+  WasmVmIntegration *clone() override { return new DummyIntegration{}; }
+  void error(std::string_view message) override {
+    std::cout << "ERROR from integration: " << message << std::endl;
+    error_message_ = message;
+  }
+  bool getNullVmFunction(std::string_view function_name, bool returns_word, int number_of_arguments,
+                         NullPlugin *plugin, void *ptr_to_function_return) override {
+    return false;
+  };
+  std::string error_message_;
+};
+
+class TestVM : public testing::TestWithParam<std::string> {
+public:
+  std::unique_ptr<proxy_wasm::WasmVm> vm_;
+
+  TestVM() : integration_(new DummyIntegration{}) {
+    runtime_ = GetParam();
+    if (runtime_ == "") {
+      EXPECT_TRUE(false) << "runtime must not be empty";
```

Comment:
What about NullVm? Existing tests in Envoy tests the same code with NullVM and all "real" WasmVMs runtimes.

Reply:
Can I make the support of NullVM in this test TODO?

Reply:
I see NullVm is already tested in `null_vm_test.cc` though it's a minimal

Reply:
Sure.